WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -343,24 +348,7 @@ public SendControlViewModel(WalletViewModel walletViewModel, bool isTransactionB
 
 					BuildTransactionResult result = await Task.Run(() => Global.WalletService.BuildTransaction(Password, intent, feeStrategy, allowUnconfirmed: true, allowedInputs: selectedCoinReferences));
 
-					if (IsTransactionBuilder)
-					{
-						var txviewer = IoC.Get<IShell>().Documents?.OfType<TransactionViewerViewModel>()?.FirstOrDefault(x => x.Wallet.Id == Wallet.Id);
-						if (txviewer is null)
-						{
-							txviewer = new TransactionViewerViewModel(Wallet);
-							IoC.Get<IShell>().AddDocument(txviewer);
-						}
-						IoC.Get<IShell>().Select(txviewer);
-
-						txviewer.Update(result);
-
-						ResetUi();
-
-						NotificationHelpers.Success("Transaction is successfully built!", "");
-
-						return;
-					}
+					await DoAfterBuildTransaction(result);
```

Comment:
@lontivero what is your opinion is this legit here? 
The first implementation is an `async Task`: 
https://github.com/zkSNACKs/WalletWasabi/pull/2990/files/87b5e3506209a4e28c7c59e5ff98b1cbd554a668..5d8cf0189768eb8b8e36492a2ddbb3c55b5d0c2f#diff-bc431b4bc31acb6d49cd62c6288310e7R24
The second is a `Task`
https://github.com/zkSNACKs/WalletWasabi/pull/2990/files/87b5e3506209a4e28c7c59e5ff98b1cbd554a668..5d8cf0189768eb8b8e36492a2ddbb3c55b5d0c2f#diff-36af77964c017c1b6ffcbf89774d0aacR22

I am using the same code to call the abstract method.

Reply:
iv done that before and I dont see any issues, you have 2 options, either dont add `async` and return CompletedTask,

or declare an `async` and await CompletedTask.

Reply:
You are awaiting a task as usual, that's okay. The difference in the `DoAfterBuildTransaction` is that given there are no `await`s then dotnet doesn't need to create an state machine for the continuations, that's all. This method is defined as `abstract Task` and then if not awaits are needed you have to return a `CompletedTask` and `FromException` as you do. This makes sense to me.